### PR TITLE
Fix build in latest Android Studio

### LINKED ISSUE #78: missing "&" in condition
in

640gAndroidUploader/app/src/main/java/info/nightscout/android/medtronic/MainActivity.java

line 489 

if (pump != null & pump.isValid()) {

should be 

if (pump != null && pump.isValid()) {

### PR DESCRIPTION
**Pull request by rahim**
This is the set of changes I needed to make to get the app to build and run locally with the latest Android Studio (2.2.2 on OSX, JDK1.8).

I'm not an Android dev so it's definitely worth some 👀  to ensure I've not done something stupid!

Resolves #78, #52

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -117,7 +117,7 @@ task zipalignRelease << {
 
 release {
     tagTemplate = 'v${version}'
-    buildTasks = ['assembleRelease']
+    buildTasks = ['build']
```
Comment (rahim) [inline app/build.gradle]: @pazaan this seemed strange/suspicious, I presume it works for you? I can see that `assembleRelease` task in the Gradle panel in the IDE. Without this change attempting to even init Gradle when the project is open fails unable to find the task.

**Comment by pazaan** [comment]
Thanks for the PR, @rahim. There were a few things that meant I couldn't directly accept the PR, the main one being that some class names had been changed. I'm happy to discuss convention on naming, but submitting a PR to fix a specific issue that modifies interfaces muddies the water a bit.

I've just committed some of your changes (and credited you). If you want to clean up this PR, I'd be happy to merge it in to make sure you get GitHub credit as well :)

**Comment by volkerrichert** [comment]
Is this PR still needed or does the current code compile on Mac (i think pazzan uses a MAC), Windows (like me) and Linux??

**Comment by pazaan** [comment]
This PR isn't required. I just left it open so that @rahim could make any changes to get GitHub credit, but he hasn't responded, so let's just close it out.

**Comment by rahim** [comment]
:) Sorry @pazaan - I think your earlier message was lost in the general noise of my Github feeds, I'd just assumed there was nothing of value here!

There was never any need to worry about credit, thanks for the mention!

**Pull request closed without merging**